airflow-core/src/airflow/utils/log/file_task_handler.py

```diff
@@ -482,6 +482,15 @@ def read(
         """
         if try_number is None:
             try_number = task_instance.try_number
+
+        if task_instance.state == TaskInstanceState.SKIPPED:
```

Comment:
@pierrejeambrun / @kaxil 

Unfortunately this handling as it is now causes another bug as regression.

It is "correct" if a short circuit or SkipMixIn really skipped the execution and the try_number is == 0 ... but there are also the cases where a task is actually executed and returns a AirflowSkipPException e.g. via s skip return code.

Example task:
![image](https://github.com/user-attachments/assets/b306a9f6-ff6e-44ad-a7d6-2d2d858254f1)

Then logs are actually existing and task has try_number >0 but logs are not showing:
![image](https://github.com/user-attachments/assets/14badd7e-9685-4678-a3c1-dedcf8b77ac9)

Reply:
Arf! My first draft was
```
        if try_number == 0 and task_instance.state == TaskInstanceState.SKIPPED:
```
And i then changed it, I didn't have this case in mind, I guess that would fix it.

Reply:
Thanks!